atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/AbstractDbWriteTable.java

```diff
@@ -133,11 +135,18 @@ public void delete(List<Entry<Cell, Long>> entries) {
 
         String prefixedTableName = prefixedTableNames.get(tableRef);
         conns.get().updateManyUnregisteredQuery(" /* DELETE_ONE (" + prefixedTableName + ") */ "
-                + " DELETE /*+ INDEX(m pk_" + prefixedTableName + ") */ "
+                + " DELETE /*+ INDEX(m " + getPrimaryKeyConstraintName(prefixedTableName) + ") */ "
                 + " FROM " + prefixedTableName + " m "
                 + " WHERE m.row_name = ? "
                 + "  AND m.col_name = ? "
                 + "  AND m.ts = ?",
                 args);
     }
+
+    private String getPrimaryKeyConstraintName(String tableName) {
+        if (config.type().equals(OracleDdlConfig.TYPE)) {
+            return OraclePrimaryKeyConstraintNames.get(tableName);
```

Comment:
Actually feels like `getPrimaryKeyConstraintName ` should be a protected method implemented differently by `OracleWriteTable` and `PostgresWriteTable` to handle their respective index/constraint limitations independently.